File: typescript/src/main/vsdk/toolkit/raycasting/bidirectionalRaytracing/LeSpar.ts
import { RadianceMethod } from "../../scene/RadianceMethod";
import { Spar } from "./Spar";
import { SparConfig } from "./SparConfig";
import { SparPathGroup } from "./SparPathGroup";

export class LeSpar extends Spar {
  public override init(sparConfig: SparConfig, radianceMethod: RadianceMethod | null): void {
    super.init(sparConfig, radianceMethod);

    if (sparConfig.baseConfig !== null && sparConfig.baseConfig.doLe !== 0) {
      this.parseAndInit(SparPathGroup.DISJOINT_GROUP, sparConfig.baseConfig.leRegExp);
    }

    if (sparConfig.baseConfig !== null && sparConfig.baseConfig.doWeighted !== 0) {
      this.parseAndInit(SparPathGroup.LD_GROUP, sparConfig.baseConfig.wleRegExp);
      this.m_sparList[SparPathGroup.LD_GROUP].add(sparConfig.ldSpar as Spar);
    }
  }
}

File: java/base/src/main/java/vsdk/toolkit/raycasting/bidirectionalRaytracing/LeSpar.java
package vsdk.toolkit.raycasting.bidirectionalRaytracing;

import vsdk.toolkit.scene.RadianceMethod;

/**
Le Spar : Uses emission ase stored radiance. Allows sampling of
all bidirectional paths
*/
public final class LeSpar extends Spar {
    @Override
    public void init(SparConfig sparConfig, RadianceMethod radianceMethod) {
        super.init(sparConfig, radianceMethod);

        // Disjoint path group for BPT
        if ( sparConfig.baseConfig.doLe != 0 ) {
            parseAndInit(SparPathGroup.DISJOINT_GROUP, sparConfig.baseConfig.leRegExp);
        }

        if ( sparConfig.baseConfig.doWeighted != 0 ) {
            parseAndInit(SparPathGroup.LD_GROUP, sparConfig.baseConfig.wleRegExp);
            m_sparList[SparPathGroup.LD_GROUP].add(sparConfig.ldSpar);
        }
    }
}
